evalml/tests/automl_tests/test_automl.py

```diff
@@ -2443,70 +2282,13 @@ def clone(self):
     )
 
 
-def _get_first_stacked_classifier_no(model_families=None):
-    """Gets the number of iterations necessary before the stacked ensemble will be used."""
-    num_classifiers = len(
-        get_estimators(ProblemTypes.BINARY, model_families=model_families)
-    )
-    # Baseline + first batch + each pipeline iteration (5 is current default pipelines_per_batch) + 1
-    return 1 + num_classifiers + num_classifiers * 5 + 1
-
-
-@pytest.mark.parametrize(
-    "max_iterations",
-    [
-        None,
-        1,
-        8,
-        10,
-        _get_first_stacked_classifier_no(),
-        _get_first_stacked_classifier_no() + 2,
-    ],
-)
-@pytest.mark.parametrize("use_ensembling", [True, False])
-def test_max_iteration_works_with_stacked_ensemble(
-    max_iterations, use_ensembling, AutoMLTestEnv, X_y_binary, caplog
-):
-    X, y = X_y_binary
-
-    automl = AutoMLSearch(
-        X_train=X,
-        y_train=y,
-        problem_type="binary",
-        max_iterations=max_iterations,
-        objective="Log Loss Binary",
-        optimize_thresholds=False,
-        ensembling=use_ensembling,
-        verbose=True,
-    )
-    env = AutoMLTestEnv("binary")
-    with env.test_context(score_return_value={"Log Loss Binary": 0.8}):
-        automl.search()
-    # every nth batch a stacked ensemble will be trained
-    if max_iterations is None:
-        max_iterations = 5  # Default value for max_iterations
-
-    pipeline_names = automl.rankings["pipeline_name"]
-    if max_iterations < _get_first_stacked_classifier_no():
-        assert not pipeline_names.str.contains("Ensemble").any()
-    elif use_ensembling:
-        assert pipeline_names.str.contains("Ensemble").any()
-        assert (
-            f"Ensembling will run at the {_get_first_stacked_classifier_no()} iteration"
-            in caplog.text
-        )
-
-    else:
-        assert not pipeline_names.str.contains("Ensemble").any()
-
-
-@pytest.mark.parametrize("max_batches", [None, 1, 5, 8, 9, 10, 12, 20])
-@pytest.mark.parametrize("use_ensembling", [True, False])
+@pytest.mark.parametrize("max_batches", [1, 5, 10])
 @pytest.mark.parametrize("problem_type", [ProblemTypes.BINARY, ProblemTypes.REGRESSION])
+@pytest.mark.parametrize("_automl_algorithm", ["iterative", "default"])
```

Comment:
I made the choice to test using both algorithms here but this can benefit from `MockAlgorithm` as well.